File: BackEnd/src/main/java/com/unosquare/admin_core/back_end/entity/Employee.java
package com.unosquare.admin_core.back_end.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

@NoArgsConstructor
@Entity
@Data
@Table(name = "Employee")
@ToString
public class Employee {

    @Id
    @SequenceGenerator(name="employeeSeq",sequenceName="employee_employee_id_seq", allocationSize = 1)
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator="employeeSeq")
    @Column(name = "employee_id", unique = true, insertable = false, nullable = false)
    private int employeeId;

    private String forename;
    private String surname;
    private String email;

    @Column(name = "total_holidays")
    private int totalHolidays;

    @Column(name = "start_date")
    private LocalDate startDate;

    @OneToOne(cascade=CascadeType.DETACH)
    @JoinColumn(name = "country_id")
    private Country country;

    private String password;

    @ManyToOne(cascade=CascadeType.DETACH)
    @JoinColumn(name = "employee_role_id")
    private EmployeeRole employeeRole;

    @ManyToOne(cascade=CascadeType.DETACH)
    @JoinColumn(name = "employee_status_id")
    private EmployeeStatus employeeStatus;

    @OneToMany(mappedBy = "employee")
    private Set<Contract> contracts = new HashSet();

    @OneToMany(mappedBy = "employee")
    private Set<Event> events = new HashSet<>();

    public Employee(int employeeId){
        this.employeeId = employeeId;
    }
}

File: v2/AdminCore.DAL/Models/Employee.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminCore.DAL.Models
{
  [Table("employee")]
  public class Employee : ISoftDeletable
  {  
    [Key]
    [Column("employee_id")]
    public int EmployeeId { get; set; }

    [Column("country_id")]
    public int CountryId { get; set; }

    [Column("email")]
    public string Email { get; set; }

    [Column("employee_role_id")]
    public int EmployeeRoleId { get; set; }

    [Column("employee_status_id")]
    public int EmployeeStatusId { get; set; }

    [StringLength(50)]
    [Column("forename")]
    public string Forename { get; set; }

    [StringLength(50)]
    [Column("password")]
    public string Password { get; set; }

    [Column("start_date")]
    public DateTime StartDate { get; set; }

    [StringLength(50)]
    [Column("surname")]
    public string Surname { get; set; }

    [Column("total_holidays")]
    public int TotalHolidays { get; set; }

    [ForeignKey("CountryId")]
    public virtual Country Country { get; set; }
    
    [ForeignKey("EmployeeRoleId")]
    public virtual EmployeeRole EmployeeRole { get; set; }
    
    [ForeignKey("EmployeeStatusId")]
    public virtual EmployeeStatus EmployeeStatus { get; set; }
  }
}